drivers/platform/maxim/maxim_uart.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************//**
+ *   @file   maxim/maxim_uart.h
+ *   @brief  Header file of UART driver.
+ *   @author Ciprian Regus (ciprian.regus@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef _MAXIM_UART_H_
+#define _MAXIM_UART_H_
+
+#include "uart.h"
+#include "no-os/irq.h"
+
+#define N_PORTS 2
+
+/**
+ * @struct maxim_uart_desc
+ * @brief Specific descriptor that contains the maxim uart config state
+ */
+struct maxim_uart_desc {
+	uint32_t port;
+	uart_cfg_t *maxim_desc;
+};
+
+/**
+ * @struct maxim_uart_init_param
+ * @brief maxim platform specific UART descriptor
+ */
+struct maxim_uart_init_param {
+	uint32_t port;
+	uart_parity_t parity;
```

Comment:
size, parity and stop configurations already exist in uart.h - they are general for all UARTs, regardless of manufacturer/implementation.
https://github.com/analogdevicesinc/no-OS/blob/master/include/no-os/uart.h#L105-L111

You should only have in maxim_uart_init_param settings that are Maxim specific only.

Reply:
I no longer use that struct for init and will delete it.